test/functional/tests/ci/test_recovery.py

```diff
@@ -0,0 +1,227 @@
+#
+# Copyright(c) 2019-2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+from os.path import basename, realpath
+
+import pytest
+
+from api.cas import casadm
+from api.cas.cache_config import CacheMode
+from core.test_run import TestRun
+from storage_devices.disk import DiskTypeLowerThan, DiskTypeSet, DiskType, Disk
+from test_tools.dd import Dd
+from test_tools.disk_utils import Filesystem, unmount, mount
+from test_tools.fs_utils import check_if_file_exists
+from test_utils.filesystem.file import File
+from test_utils.os_utils import sync
+from test_utils.output import CmdException
+from test_utils.size import Size, Unit
+
+mount_point = "/mnt/cas"
+
+
+@pytest.mark.CI
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_recover_cache():
```

Comment:
I would rename the tests to better reflect the difference:
this one: test_recover_cache_verify_core()
the next one: test_recover_cache_verify_exp_obj()